pkg/clients/database/cosmosdb/account.go

```diff
@@ -0,0 +1,231 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cosmosdb
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2015-04-08/documentdb"
+	"github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2015-04-08/documentdb/documentdbapi"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/pkg/errors"
+
+	"github.com/crossplane/provider-azure/apis/database/v1beta1"
+	azure "github.com/crossplane/provider-azure/pkg/clients"
+)
+
+// A AccountClient handles CRUD operations for Azure CosmosDB Accounts.
+type AccountClient documentdbapi.DatabaseAccountsClientAPI
+
+// NewDatabaseAccountClient create Azure DatabaseAccountsClient using provided credentials data
+func NewDatabaseAccountClient(credentials []byte) (AccountClient, error) {
+	creds := &azure.Credentials{}
+	if err := json.Unmarshal(credentials, creds); err != nil {
+		return nil, errors.Wrap(err, "cannot unmarshal Azure client secret data")
+	}
+
+	config := auth.NewClientCredentialsConfig(creds.ClientID, creds.ClientSecret, creds.TenantID)
+	config.AADEndpoint = creds.ActiveDirectoryEndpointURL
+	config.Resource = creds.ResourceManagerEndpointURL
+
+	authorizer, err := config.Authorizer()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get authorizer from config: %+v", err)
+	}
+
+	client := documentdb.NewDatabaseAccountsClient(creds.SubscriptionID)
+	client.Authorizer = authorizer
+
+	if err := client.AddToUserAgent(azure.UserAgent); err != nil {
+		return nil, errors.Wrap(err, "cannot add to Azure client user agent")
+	}
+
+	return client, nil
+}
+
+// ToDatabaseAccountCreate from DatabaseAccountSpec
+func ToDatabaseAccountCreate(s *v1beta1.CosmosDBAccountSpec) documentdb.DatabaseAccountCreateUpdateParameters {
+	if s == nil {
+		return documentdb.DatabaseAccountCreateUpdateParameters{}
+	}
+
+	//DatabaseAccountProperties
+	return documentdb.DatabaseAccountCreateUpdateParameters{
+		Kind:                                  s.ForProvider.Kind,
+		Location:                              azure.ToStringPtr(s.ForProvider.Location),
+		Tags:                                  azure.ToStringPtrMap(s.ForProvider.Tags),
+		DatabaseAccountCreateUpdateProperties: toDatabaseProperties(&s.ForProvider.Properties),
+	}
+}
+
+// ToDatabaseAccountUpdate from DatabaseAccountSpec
+func ToDatabaseAccountUpdate(s *v1beta1.CosmosDBAccountSpec) documentdb.DatabaseAccountCreateUpdateParameters {
+	return ToDatabaseAccountCreate(s)
```

Comment:
Maybe these two should just be `ToDatabaseCreateOrUpdate`? I don't feel strongly either way, though.